File: frontend/src/types/warehouseInventory.ts
export interface WarehouseInventoryItem {
  id: number;
  warehouse_id: number;
  product_id: number;
  product_name: string;
  product_sku: string;
  category_name: string;
  warehouse_name: string;
  warehouse_code: string;
  quantity: number;
  minimum_quantity: number;
  maximum_quantity: number;
  is_low_stock: boolean;
  updated_at: string;
}

File: backend/app/schemas/warehouse_inventory.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, ConfigDict


class WarehouseInventoryUpdate(BaseModel):
    minimum_quantity: Optional[int] = None
    maximum_quantity: Optional[int] = None


class WarehouseInventoryOut(BaseModel):
    id: int
    warehouse_id: int
    product_id: int
    product_name: str
    product_sku: str
    category_name: str
    warehouse_name: str
    warehouse_code: str
    quantity: int
    minimum_quantity: int
    maximum_quantity: int
    is_low_stock: bool
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)
